src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/ICodeGenerationArrayTypeSymbol.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CodeGeneration;
+
+/// <summary>
+/// This interface provides a subset of <see cref="IArrayTypeSymbol"/> for use in code generation scenarios.
+/// </summary>
+internal interface ICodeGenerationArrayTypeSymbol : ICodeGenerationTypeSymbol
+{
```

Comment:
this is also moving us to a world where we have *less* code sharing, instead of more.  Which is an anti-goal for roslyn, especially in the IDE.

Reply:
➡️ This change is part of prerequisite pull request https://github.com/dotnet/roslyn/pull/73675. Resolving this comment as off-topic.